### PR TITLE
CP-32846 drop legacy ssl support

### PR DESCRIPTION
**Pull request by lippirk**
Use TLS v1.2 instead of deprecated SSL

Signed-off-by: lippirk <ben.anson@citrix.com>

### CODE REVIEW

**Comment by edwintorok** [comment]
Travis is out of date here, try updating it based on a working repo, such as xapi-idl (just keep the matrix for all the opam packages)

File: .travis.yml [hunk 1]
```diff
@@ -1,25 +1,12 @@
...
   global:
-    - OCAML_VERSION=4.07
     - PINS="xen-api-client:. xen-api-client-lwt:. xen-api-client-async:."
     - PACKAGE=xen-api-client
-    - DISTRO="debian-9"
-  matrix:
-    - BASE_REMOTE=git://github.com/xapi-project/xs-opam \
-        POST_INSTALL_HOOK="env TRAVIS=$TRAVIS TRAVIS_JOB_ID=$TRAVIS_JOB_ID bash -ex coverage.sh"
-    - PACKAGE=xen-api-client-async BASE_REMOTE=git://github.com/xapi-project/xs-opam POST_INSTALL_HOOK="make async-examples"
-    - PACKAGE=xen-api-client-lwt BASE_REMOTE=git://github.com/xapi-project/xs-opam POST_INSTALL_HOOL="make lwt-examples"
```
Comment (psafont) [inline .travis.yml]: Please keep these two jobs (even though the lwt was broken because of a typoe in `INSTALL_HOOL`)

File: .travis.yml [hunk 2]
```diff
@@ -1,25 +1,15 @@
...
-    - EXTRA_REMOTES=git://github.com/xapi-project/xs-opam
-matrix:
-  fast_finish: true
-  allow_failures:
-    - env: EXTRA_REMOTES=git://github.com/xapi-project/xs-opam
+    - PACKAGE=xen-api-client-async POST_INSTALL_HOOK=
+    - PACKAGE=xen-api-client-lwt POST_INSTALL_HOOL="make lwt-examples"
```
Comment (edwintorok) [inline .travis.yml]: I think this should say `POST_INSTALL_HOOK="make lwt-examples"`, and similarly for the async one `make async-examples`

**Comment by lindig** [comment]
The Travis job for lwt still fails:
```
+ make lwt-examples
dune build @lwt_examples/examples
File "lwt_examples/dune", line 25, characters 4-19:
25 |     cohttp-lwt-unix
         ^^^^^^^^^^^^^^^
Error: Library "cohttp-lwt-unix" not found.
Hint: try: dune external-lib-deps --missing @lwt_examples/examples
make: *** [Makefile:17: lwt-examples] Error 1
'unset TESTS; OPAMYES=1 export OPAMYES; set -uex; make lwt-examples' exited 2. Terminating with 2
```

**Comment by psafont** [comment]
The dependency on`cohttp-lwt` is missing from `xen-api-client-lwt.opam`

**Comment by lindig** [comment]
Looks good to me. Can we squash all the Travis commits such that we are left with two commits?

File: lwt/data_channel.ml [hunk 3]
```diff
@@ -67,7 +67,7 @@ let of_fd fd ~seekable = (if seekable then of_seekable_fd else of_unseekable_fd)
 
 let sslctx =
   Ssl.init ();
-  Ssl.create_context Ssl.SSLv23 Ssl.Client_context
+  Ssl.create_context Ssl.TLSv1_2 Ssl.Client_context
```
Comment (psafont) [inline lwt/data_channel.ml]: Is there a mode that can use TLS 1.3 as well as 1.2? `SSLv23` does that but also fall back even further if OpenSSL is compiled with support for legacy protocols

Comment (psafont) [reply lwt/data_channel.ml]: nevermind, openssl 1.0.2 doesn't support tls 1.3 :(

**Pull request merged**